### DISCUSSION TITLE
Copilot.vim actually works with non-Neovim editors

### DISCUSSION THREAD

**Discussion by panozzaj** [community/community#49733] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

Hi, thanks for the Vim plugin, loving using it! I wanted to report a small documentation improvement opportunity.

The repo description of https://github.com/github/copilot.vim says "Neovim plugin for GitHub Copilot". My preferred editor is MacVim. As a result of seeing this and internet search seemingly confirming that it's only usable on Neovim, I went down the path of getting Neovim working to try out Copilot.

Then later, when I was working in Macvim I surprisingly got some completions! After reading the repo documentation more thoroughly, it says that it works for "Neovim or the latest patch of Vim (9.0.0185 or newer)."

I was wondering if this would be an appropriate place to ask someone to clarify the repo description, as I spent some time getting things working in Neovim that was not totally needed. Also, most of the documentation online (articles, etc.) seems to refer to Neovim only, and this could help improve that situation.

My Macvim version:

```
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Sep 15 2022 19:27:32)
macOS version - x86_64
Included patches: 1-472
```

Thanks!

**Comment by holstvoogd** [comment]
Funny, for me it does not work at all in any version or variant of vim. Anything in particular you did to get it working on neovim? Or just, get neovim working instead of you macvim? I can't even run the setup from nvim 😂

**Reply by panozzaj** [reply]
Hmm, not sure, seems like a different challenge than what I reported. I followed (more or less) the instructions at https://github.com/github/copilot.vim. I [got neovim to look at my `~/.vimrc`](https://vi.stackexchange.com/questions/12794/how-to-share-config-between-vim-and-neovim/15548#15548) since I'm using Pathogen + submodules to manage plugins. Then ran the Copilot setup in Neovim. I didn't really change anything for MacVim to see it. If you can't get `:Copilot setup` to execute, I'd suspect neovim is not finding the plugin.

**Comment by jmiller-rise8** [comment]
Probably depends on whomever is going to maintain this repository into the future and the plans that the larger Copilot IDE team. Not sure I’d get comfortable using this on MacVim.

**Reply by ychin** [reply]
(MacVim maintainer here) I have been using Copilot here and there to try it out and it works. The [plugin](https://github.com/github/copilot.vim) explicitly says it works in Vim in the README. My understanding is that the key feature that it needs is virtual texts, which was implement in Vim 9.

The issue here is mostly the marketing pages have not been updated to reflect that (e.g. https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim) (and also the repo description, I guess). It was true when the plugin was written that only Neovim was supported, but it's not true anymore, and so documentation should be updated. Right now the only way you would find out that Vim works just fine is by going to the repo itself and reading the README. It personally took me a couple double-takes before I figured out that the docs/marketing page was wrong and the plugin actually works.

**Reply by ychin** [reply]
Oh actually seems like the docs https://github.com/github/docs/ are public and allow pull requests. Maybe I'll just submit one. No point in whining here.

**Reply by jmiller-rise8** [reply]
@ychin 
> Oh actually seems like the docs https://github.com/github/docs/ are public and allow pull requests. Maybe I'll just submit one. No point in whining here.

Unfortunately it does not currently accept pull requests nor do we know if it will moving forward.

> The [plugin](https://github.com/github/copilot.vim) explicitly says it works in Vim in the README. My understanding is that the key feature that it needs is virtual texts, which was implement in Vim 9.

I'm aware. The point I was making is that the new maintainer of the project and Copilot IDE team will probably be looking to find feature parity across IDEs.

**Reply by ychin** [reply]
They do take pull requests! My pull request was just merged: [24625](https://github.com/github/docs/pull/24625).

You can now view it here: https://docs.github.com/en/copilot/getting-started-with-github-copilot?tool=vimneovim

**Comment by ychin** [comment, marked as answer]
I mentioned in a reply to another answer, by I had a PR that was now merged (https://github.com/github/docs/pull/24625), so the official docs now says "Vim/Neovim" instead of just "Neovim". You can see it here: https://docs.github.com/en/copilot/getting-started-with-github-copilot?tool=vimneovim.

As for the repo description of https://github.com/github/copilot.vim, I think only @tpope can change that. The README already mentions Vim anyway.

**Reply by panozzaj** [reply]
OK, thanks for doing this! I don't have a way of reaching out to Tim through the repo, so the @ mentions you have here and on the docs PR would be best. I'm going to mark this as the answer / mark this issue as resolved. Thanks again!
